### ISSUE TITLE
Show Left Area toggled command in palette not updated

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#4524]
The toggled state of the "Show Left Area" command is not shown/updated in a command palette when it should be. For example, having the command palette open, click the "Show Left Area" option to untoggle it. Then click on the command palette tab to re-open the left area. For me, the "Show Left Area" command remains untoggled in the command palette, but should be toggled since the left area is now open.

cf #4510

**Comment by jasongrout** [comment]
I also notice the command palette toggled status for single-document mode not updating when I toggle it from the menu with the palette closed.

**Comment by ian-r-rose** [comment]
There are some places where we should be more aggressive about when to rerender the command palette. This affects not just the toggled state, but also which commands are enabled/disabled. I have caught some of them, but clearly there are some that I have missed. Can we use this issue as a place for collecting those as we see them?

**Comment by jasongrout** [comment]
Sure, feel free to update the issue title to be more general, then.

**Comment by jasongrout** [comment]
Enlightening conversation on gitter, starting at https://gitter.im/jupyterlab/jupyterlab?at=5aecc3c603cafa797b297bbc

tldr; any updates that cause command state to update should result in calling the commandNotifyChange function, exactly as the docs say: https://github.com/phosphorjs/phosphor/blob/master/packages/commands/src/index.ts#L137-L156

**Comment by jasongrout** [comment]
Thanks again @ian-r-rose!
